tests/src/Simple/Threading/Threading.cs

```diff
@@ -391,6 +394,25 @@ public static void SemaphoreTest()
     }
 
     [Fact]
+    public static void Reproduce3616()
+    {
+        AutoResetEvent resetEvent = new AutoResetEvent(false);
+        Thread thread = new Thread(() => {
+            var i = 0;
+            var tickCount = Environment.TickCount;
+            while(Environment.TickCount < tickCount + 1000) {
+                i++;
+            }
+            resetEvent.Set();
+            resetEvent.Set();
+        });
+
+        thread.Start();
```

Comment:
Would be good to set `thread.IsBackground = true` by default for threads before starting them, unless there is a reason it should not be a background thread, so that they won't block the process from terminating if something goes wrong and the thread doesn't terminate. The test should instead fail at the `Join` if the thread does not terminate.